### ISSUE TITLE
AmpereOne A192-32X

### ISSUE THREAD

**Issue by ThomasKaiser** [hrw/arm-socs-table#111]
System name: Super Server MegaDC ARS-211ME-FNR
Vendor name: Supermicro
SoC name: AmpereOne A192-32X
SoC vendor name: Ampere
URL to product page: [SoC family and SKUs](https://amperecomputing.com/briefs/ampereone-family-product-brief) / [Server system](https://www.supermicro.com/en/products/system/megadc/2u/ars-211me-fnr)
Kernel version (if known): 6.8.0

```processor	: 0
BogoMIPS	: 2000.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 sm3 sm4 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint i8mm bf16 rng bti ecv
CPU implementer	: 0xc0
CPU architecture: 8
CPU variant	: 0x0
CPU part	: 0xac4
CPU revision	: 0
```

([full `/proc/cpuinfo` showing all 192 cores](https://github.com/ThomasKaiser/sbc-bench/blob/master/results/cpuinfo/Ampere-A192-32X-6.8.0.cpuinfo))

Please note that this SoC uses the newer Ampere-1a cores (`0xc0`/`0xac4` r0p0). Data collected by [geerlingguy](https://github.com/geerlingguy/sbc-reviews/issues/52) as part of an [sbc-bench run](https://github.com/ThomasKaiser/sbc-bench/issues/105).

**Comment by hrw** [comment]
Hm. Ampere1a should have MTE as well.

**Comment by ThomasKaiser** [comment]
> Ampere1a should have MTE as well.

Isn't that an ARMv9 feature?

**Comment by hrw** [comment]
https://community.amperecomputing.com/t/weekend-read-memory-tagging-in-ampere-one/963 shows that MTE is present.

```
mte:
  archv: 8.5
  feat: FEAT_MTE
  name: Memory Tagging Extension
  note: v8.5 optional
mte3:
  archv: 8.5
  feat: FEAT_MTE3
  name: MTE Asymmetric Fault Handling
  note: v8.5 optional, v8.7 mandatory
```

**Comment by hrw** [comment]
@bexcran

**Comment by bexcran** [comment]
Yup I’ve seen this and I’m working on it right now. Trying to figure out why FEAT_MTE isn’t showing up.

**Comment by ThomasKaiser** [comment]
> Trying to figure out why FEAT_MTE isn’t showing up.

Maybe related to `CONFIG_ARM64_MTE` not set appropriately in default Ubuntu kernel config?

**Comment by bexcran** [comment]
I’d hope it doesn’t need that and instead just reports the features the CPU supports.

**Comment by bexcran** [comment]
It looks like MTE (and other features) are disabled by default in the firmware, and you may need access to Customer Connect in order to get the documentation to enable them.

**Comment by hrw** [comment]
@geerlingguy ;D

**Comment by geerlingguy** [comment]
@bexcran what is Customer Connect? Is that through Supermicro, or Ampere?

**Comment by bexcran** [comment]
> @bexcran what is Customer Connect? Is that through Supermicro, or Ampere?

It's Ampere's customer portal: https://amperecomputing.com/customer-connect which gives access to downloads and documentation which is under an NDA.

**Comment by geerlingguy** [comment]
On my machine as shipped:

```
ubuntu@ubuntu:/opt/top500/tmp$ grep CONFIG_ARM64_MTE /boot/config-6.8.0*
/boot/config-6.8.0-39-generic-64k:CONFIG_ARM64_MTE=y
/boot/config-6.8.0-47-generic:CONFIG_ARM64_MTE=y
```

**Comment by abajk** [comment]
Related link: https://finkmartin.com/posts/arm-mte-ampere-1a/
